students/anbaopham/Lesson_4/Lesson_4_Trigrams.py

```diff
@@ -0,0 +1,55 @@
+import random
+
+#words = ("I wish I may I wish I might").split()
+import string
+with open('sherlock_small.txt', 'r') as file:
+    text_file = file.read().replace('\n', '')
+
+symbols ='-.()--'
+
+words = text_file.translate({ord("-"): " ",ord("("): "",
+ord(")"): "",ord("."): "",ord(","): ""})
+words = words.split()
+
+def check_dictionary(trigrams, key):
+    if key in trigrams:
+        return True
+    else:
+        return False
+def get_value(words, pair):
+    value_list = []
+    for i in range(len(words)-2):
+        if words[i] == pair[0] and words[i:i+2] == pair:
+            value_list.append(words[i+len(pair)])
+    return value_list
+
+
+
+def build_trigrams(words):
+    trigrams = {}
+    value =[]
+    random_choice = random.choice(words)
+    n =words.index(random_choice)
+    for i in range(n,len(words)-2):
+        pair = words[i:i+2]
+        key = " ".join(pair)
+        follower = words[i+2]
+
+        if check_dictionary(trigrams, key) is False:
+            trigrams[key] = follower
+    for key in trigrams:
+        pair = key.split()
+        trigrams[key] = get_value(words, pair)
+    print(random_choice)
+    print(n)
+    display_dictionary(trigrams)
+    return trigrams
+
+
+def display_dictionary(trigrams):
+    for x,y in trigrams.items():
+        print(x,y)
+
+
+
+build_trigrams(words)
```

Comment:
An - this isn't producing the expected results.  Is this code complete?

Reply:
I think it worked because it worked with "words" in line #3.  If we don't read "sherlock_small.txt", this code worked with "words" in line 3.
Did I miss something?

Reply:
The output should be a created 'story' with sentences created by the functions in your code.

Reply:
If I run line #3, it showed as expected.  Can you give please more provide more detail how to improve my code?

Reply:
The user should be able to pass in a file, like sherlock_small.txt, and the output should be a new, randomly generated short story.  Does that make sense.  I'm going to merge this PR, so please submit any new or edited code to a new PR for further review.